src/test/ui/derives/issue-97343.rs

```diff
@@ -0,0 +1,8 @@
+use std::fmt::Debug;
+
+#[derive(Debug)]
+pub struct Irrelevant<Irrelevant> { //~ ERROR type arguments are not allowed for this type
```

Comment:
Huh, this error message is very confusing imo. I guess that's not necessarily the purpose of this PR. 
r=me anyways.

Reply:
Leaving https://github.com/rust-lang/rust/issues/97343 open to both address that (I think a new error code is warranted) and potentially make the macro smarter to make this code work in the first place.